modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/IO.kt

```diff
@@ -163,6 +166,15 @@ sealed class IO<out A> : IOOf<A> {
     // Pure can be replaced by its value
     override fun <B> map(f: (A) -> B): IO<B> = Suspend { Pure(f(a)) }
 
+    // Pure can be replaced by its value
+    override fun <B> mapFilter(f: (A) -> Option<B>): IO<B> =
+      Suspend {
+        f(a).fold(
+          { RaiseError(IllegalArgumentException("IO execution yield invalid result after mapFilter")) },
```

Comment:
This is not ok. We can;t have unsafe operations like this just because we can raise Throwables. This makes IO unsafe and non pure because IO can't provide an empty value.

Reply:
I'm going to remove this. Thanks for feedback :+1: